SetupBLT.cmake

```diff
@@ -101,6 +101,14 @@ if (NOT BLT_LOADED)
     ################################
     set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
+
+    ################################
+    # PGI doesn't support the SYSTEM flag, which is the default for imported targets.
+    ################################
+    if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "PGI")
+        set(CMAKE_NO_SYSTEM_FROM_IMPORTED TRUE)
+    endif()
```

Comment:
This resolves the issue, but I'm not thrilled with the solution.

I'd like this fix to go inside `cmake/SetupCompilerOptions` since it is a specific fix for the PGI compiler.
However, the latter has a not that it has to be included after MPI is set up:
https://github.com/LLNL/blt/blob/113f4debe85e03f7f37b9ecd4bc0735da619dc2c/SetupBLT.cmake#L192-L196
which would seem be too late to affect our import targets.

Perhaps we should refactor the code so `SetupCompilerOptions.cmake` can be included before `SetupThirdParty.cmake` ?

Reply:
That could be a good idea but we do set some global properties in this file.

Reply:
Good news -- Looks like I'm able to move the workaround to `SetupCompilerOptions.cmake` without any refactoring. Will push an update momentarily.